libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessage.java

```diff
@@ -130,7 +130,20 @@ public void setFieldCount(int count) {
      *   of a field with the specified tag
      */
     public FIXValue valueOf(int tag) {
-        for (int i = 0; i < count; i++) {
+        return valueOf(tag, 0);
+    }
```

Comment:
I would retain the existing `valueOf` body to avoid the indirection in the most common case.